File: 49-group-anagrams/group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const hash: Map<string, string[]> = new Map();

    for (const str of strs) {
        const sorted: string = str.split("").sort().join();

        if (!hash.has(sorted)) {
            hash.set(sorted, []);
        }

        hash.get(sorted)!.push(str);
    }

    return [...hash.values()];
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        grouped_anagrams = defaultdict(list)

        for string in strs:
            key = "".join(sorted(string))
            grouped_anagrams[key].append(string)

        return list(grouped_anagrams.values())